src/screencast/wlr_screencast.c

```diff
@@ -34,6 +35,11 @@ static void wlr_frame_buffer_destroy(struct xdpw_screencast_instance *cast) {
 	}
 }
 
+static void register_callback(void *data) {
```

Comment:
Is this being added only to change the argument to a void pointer?

Reply:
Yeah, it is. I have no idea how to cast function pointers ☺️ I am open to suggestions.

Reply:
I'd say just rewrite the original function signature and add the type cast there if it is going to be regularly called by a callback. It won't affect existing functionality.

We'll lose a small amount of type safety but I don't know of a better way either.

Reply:
I just pushed a commit that removes this function and replaces its use with an actual cast to `xdpw_wlr_register_cb`. There was no need after all to change the function `xdpw_wlr_register_cb`. The cast turned out to be simple, I don't know why it didn't compile for me when I tried it before. Anyway, I am marking this as resolved - @danshick please reopen if you don't like the way it came out.

Reply:
Yeah, this looks fine! I didn't know that cast would work TBH but this is much better.